src/main/java/com/zendesk/maxwell/util/TopicInterpolator.java

```diff
@@ -0,0 +1,89 @@
+package com.zendesk.maxwell.util;
+
+import com.zendesk.maxwell.row.RowIdentity;
+import com.zendesk.maxwell.row.RowMap;
+
+/**
+ * An utility class to interpolate subscription, channel templates.
+ * <p>
+ * An input string can contains any combination of:
+ * - %{db}
+ * - %{table}
+ * - %{type}
+ */
+public class TopicInterpolator {
+	private final String inputString;
+
+	private final boolean isInterpolated;
+
+	public TopicInterpolator(final String templateString) {
+		this.inputString = templateString;
+		this.isInterpolated = templateString.contains("%{");
+	}
+
+	/**
+	 * Interpolate the input string based on {{{@link RowIdentity}}}
+	 * <p>
+	 * If your inputString contains a %{type} it will not be interpolated because we can't get type from RowIdentity.
+	 *
+	 * @param pk the rowIdentity
+	 * @return the interpollated string
+	 */
+	public String generateFromRowIdentity(RowIdentity pk) {
+		String table = pk.getTable();
+
+		if (this.isInterpolated)
+			return interpolate(pk.getDatabase(), table, null);
+		else
+			return this.inputString;
+	}
+
+	/**
+	 * Interpolate the input string based on {{{@link RowMap}}}
+	 *
+	 * @param r the rowMap
+	 * @return the interpollated string
+	 */
+	public String generateFromRowMap(RowMap r) {
+		String table = r.getTable();
+
+		String type = r.getRowType();
+
+		if (this.isInterpolated)
+			return interpolate(r.getDatabase(), table, type);
+		else
+			return this.inputString;
+	}
+
+	protected String interpolate(final String database, final String table, final String type) {
+		if (this.isInterpolated) {
+			final String typeReplacement = type != null ? type : "";
+
+
+			return this.inputString
+					.replaceAll("%\\{database\\}", cleanupIllegalCharacters(cleanupIllegalCharacters(emptyStringOnNull(database))))
+					.replaceAll("%\\{table\\}", cleanupIllegalCharacters(cleanupIllegalCharacters(emptyStringOnNull(table))))
+					.replaceAll("%\\{type\\}", cleanupIllegalCharacters(cleanupIllegalCharacters(emptyStringOnNull(typeReplacement))));
```

Comment:
- shouldn't call `cleanupIllegalCharacters` twice
- please only do this cleanup for "nats" -- kafka topic rules are different and I don't want to change behavior there.

Reply:
updated in https://github.com/zendesk/maxwell/pull/1646/commits/d6e9b0077b5add7e9385ad05c2b097e63accc208#diff-45da18462c44ee750de35aea378ff0075b32a7f16d6c688a3cdbc5bfde4c5e9eR78